src/components/modes/teacher/Settings.js

```diff
@@ -113,70 +120,28 @@ class Settings extends Component {
   };
 
   handleClose = () => {
-    const { dispatchCloseSettings } = this.props;
+    const { dispatchCloseSettings, settings, dispatchGetActions } = this.props;
+    const { selectedSpacesState } = this.state;
     dispatchCloseSettings();
-  };
 
-  handleChangeHiddenVerbs = verb => {
-    const {
-      settings: { hiddenVerbs = [] },
-    } = this.props;
-    const checked = !hiddenVerbs.includes(verb);
-    let newHiddenVerbs = [...hiddenVerbs];
-    if (checked) {
-      newHiddenVerbs.push(verb);
-    } else {
-      newHiddenVerbs = newHiddenVerbs.filter(thisVerb => thisVerb !== verb);
+    // fetch actions again if selected spaces changed
+    if (!_.isEqual(selectedSpacesState, settings.spaces)) {
+      dispatchGetActions();
     }
-    this.saveSettings({ hiddenVerbs: newHiddenVerbs });
   };
 
-  renderActionChecks = () => {
-    const {
-      verbs,
-      classes,
-      t,
-      settings: { hiddenVerbs = [] },
-    } = this.props;
+  handleOpen = () => {
+    const { dispatchOpenSettings, settings } = this.props;
+    // set initial settings state to compare on close
+    this.setState({ selectedSpacesState: settings.spaces });
 
-    const checkboxes = verbs.map(verb => {
-      const checkbox = (
-        <Checkbox
-          color="primary"
-          checked={!hiddenVerbs.includes(verb)}
-          onChange={() => this.handleChangeHiddenVerbs(verb)}
-          name={verb}
-          value={verb}
-        />
-      );
-      return (
-        <FormControlLabel
-          className={classes.checkbox}
-          control={checkbox}
-          label={verb}
-        />
-      );
-    });
-    return (
-      <>
-        <FormControl className={classes.verbForm}>
-          <Typography variant="h6" className={classes.verbTitle}>
-            {t('Include the following action verbs')}
-          </Typography>
-          {checkboxes}
-        </FormControl>
-      </>
-    );
+    dispatchOpenSettings();
   };
 
   renderModalContent() {
-    const { t, settings, activity, classes } = this.props;
+    const { t, settings, classes } = this.props;
     const { headerVisible } = settings;
 
-    if (activity) {
```

Comment:
This part is removed to avoid rendering when a setting is changed (select a space, select an action verb, ...)